server/src/main/scala/controllers/Application.scala

```diff
@@ -341,6 +341,16 @@ class Application @Inject()(implicit val config: Configuration,
     }
   }
 
+  def timed[A](description: String)(action: Action[A]): Action[A] = Action.async(action.parser) { request =>
+    val startMillis = SDate.now().millisSinceEpoch
+    val actionResponse = action(request)
+    val endMillis = SDate.now().millisSinceEpoch
+
+    log.info(s"$description (${request.uri}) took ${endMillis - startMillis}ms")
+
+    actionResponse
+  }
```

Comment:
would it be possible to write this more generically so that it can be used to time any future?

```scala
def timed[A](description: String)(future: Future[A]) {
  val startMillis = SDate.now().millisSinceEpoch
  future.onComplete({
      case _ => 
        val endMillis = SDate.now().millisSinceEpoch

       log.info(s"$description (${request.uri}) took ${endMillis - startMillis}ms")
    })
   future
}
 ```

Reply:
may not be worth it, but could be handy.